Yoast/wordpress-seo
# Adds OAuth for communication with SEMrush API.

## Context


* Introduces the OAuth integration with SEMrush and allows basic authentication and token refreshing.

## Summary


This PR can be summarized in the following changelog entry:

* Adds OAuth for communication with SEMrush API.

## Relevant technical choices:

* Please note that although the refreshing of tokens is already built-in, this hasn't been fully implemented / tested, as it relies on other endpoints being built on top of this first.

## Test instructions

This PR can be tested by following these steps:

* Ensure you've run `composer install`
* In your browser, visit the authorization link provided by SEMrush (available upon request to avoid spamming)
* Authorize SEMrush and copy the value of the `code` parameter in your address bar.
* In Postman (or a similar application), add a new POST request to `http://basic.wordpress.test/wp-json/yoast/v1/semrush/authenticate`
* In the `body` of the request, add a new value named `code` and paste the copied value from your address bar, as its value.
* Send the request. You should receive a valid response with a 200 status code and an `access_token`, `refresh_token`, `expiration` and `has_expired` property.
* Attempt to send the same request again *or* change one of the digits in the `code` value.
* You should receive an error response.

Additional
* Run the tests, ensure they pass.

## UI changes

## Documentation

## Quality assurance
